src/main/java/org/opensearch/security/dlic/rest/validation/AbstractConfigurationValidator.java

```diff
@@ -297,4 +315,26 @@ public String getMessage() {
     protected final boolean hasParams() {
         return param != null && param.length > 0;
     }
-}
\ No newline at end of file
+
+    private boolean hasNullElement(JsonNode node) {
+        if (node.isArray()) {
+            for (JsonNode jsonNode: node){
+                if(jsonNode.isNull()){
+                    return true;
+                }
+            }
+            ArrayNode arrayNode = (ArrayNode) node;
+            Iterator<JsonNode> nodeIter = arrayNode.elements();
```

Comment:
It is confusing in a single method to use `node.iterator()`, `node.elements()` and `node.fields()`. Please unify to use `node.iterator()`.

Reply:
I am not sure how should I do it, as it depends whether the `JsonNode` is `array` or `object`. I can not see any confusion as the if statements are clear enough to see the type of the JsonNode and how we are handling it.

Reply:
Won't `node.iterator()` (`for (JsonNode jsonNode: node) {}`) work both for `array` and `object`? Why is it necessary to use `node.fields()` for `object`?

Reply:
Done

Reply:
I tried to use only `node.iterator` but it cause failure in some other tests like `AuditApiActionTest`, debugging it showed that it can not handle all the objects.

Reply:
Please provide details why `node.iterator()` can not handle all objects.

Reply:
We just want to make sure that an array element does not have a null, right? This code:
```
for (JsonNode jsonNode : node) {
      if(jsonNode.isNull()){
         return true;
       }
       else {
            Iterator<JsonNode> nodeIter = jsonNode.iterator();
             while (nodeIter.hasNext()) {
                        if (hasNullElement(nodeIter.next()))
                            return true;
              }
       }
}
```
return `true` for inputs like this:
```{"enabled":true,"write_log_diffs":false,"read_watched_fields":null,"read_ignore_users":[],"write_watched_indices":[],"write_ignore_users":["kibanaserver"],"read_metadata_only":true,"write_metadata_only":false,"external_config":true,"internal_config":true}```

Because the value of `read_watched_fields` is `null`, however it is ok, since `read_watched_fields` is not an array.
What is wrong with the implementation in the last commit? I made it more clear and it is really straight forward.

Reply:
OK, I modified `hasNullElement()` method to only use node iterator in my latest commit.. Hopefully it is good now.